pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -313,13 +315,38 @@ def store_predictions(self, output, batch_idx, dataloader_idx):
 
     def on_evaluation_epoch_end(self, *args, **kwargs):
         # call the callback hook
-        if self.trainer.testing:
-            self.trainer.call_hook('on_test_epoch_end', *args, **kwargs)
-        else:
-            self.trainer.call_hook('on_validation_epoch_end', *args, **kwargs)
+        self.call_on_evaluation_epoch_end_hook()
 
         self.trainer.call_hook('on_epoch_end')
 
+    def call_on_evaluation_epoch_end_hook(self):
+        outputs = self.outputs
+        model_ref = self.trainer.get_model()
+        hook_name = "on_test_epoch_end" if self.trainer.testing else "on_validation_epoch_end"
+
+        self.trainer._reset_result_and_set_hook_fx_name(hook_name)
```

Comment:
Why not using self.trainer.call_hook with outputs directly if the outputs is present ?

Reply:
`self.trainer.call_hook` addresses `callback_hooks` and `model_hook` together, and pass the same arguments. We need to inspect both of them separately and pass `outputs` based on the signature.